File: python/exo6_3a.py
import sys
import struct

n = int(input("Nombre d'entiers: "))
tableau = []
for i in range(n) :
        tableau.append(int(input(f"Nombre {i+1}: ")))

try :
    fd = open("fichierBin", "wb")  #permission
except :
        print("Erreur lors de l'ouverture du fichier.")
        sys.exit(1)

fd.write(struct.pack(f"{n}i", *tableau))
fd.close()
print("Sauvegarde terminée dans fichierBin")

File: C/exo6_3a.c
#include <stdio.h>
#include <fcntl.h>
#include <unistd.h>

int main()
{
    int tableau[100];
    int n, i;
    int fd;
    
    printf("Nombre d'entiers: ");
    scanf("%d", &n);    
    for(i=0; i<n; i++)
    {
        printf("Nombre %d: ", i+1);
        scanf("%d", &tableau[i]);
    }

    fd=open("fichierBin", O_CREAT | O_WRONLY | O_TRUNC, 0664); //permission
    if(fd==-1)
    {
        printf("Erreur lors de l'ouverture du fichier.\n");
        exit(1);
    }
    
    write(fd, tableau, n * sizeof(int));
    close(fd);    
    printf("Sauvegarde terminée dans fichierBin\n");   
    return 0;
}